### PR TITLE
switchable caching

### PR DESCRIPTION
**Pull request by sinasohangirsc**
* More efficient isLegacy method based on first line of the sql query.
* Add switch to enable/disable caching

### CODE REVIEW

File: scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryClient.scala [hunk 1]
```diff
@@ -528,6 +537,13 @@ object BigQueryClient {
   /** Default cache directory. */
   val CACHE_DIRECTORY_DEFAULT: String = sys.props("user.dir") + "/.bigquery"
 
+  /** System property key for enabling or disabling scio bigquery caching */
+  val CACHE_ENABLED: String = "bigquery.cache.enabled"
```
Comment (ravwojdyla) [inline scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryClient.scala]: To keep it consistent `CACHE_ENABLED` -> `CACHE_ENABLED_KEY`

File: scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryClient.scala [hunk 2]
```diff
@@ -426,27 +428,34 @@ class BigQueryClient private (private val projectId: String,
...
-      case Failure(e: GoogleJsonResponseException) if isInvalidQuery(e) =>
-        // dry run with legacy syntax next
-        dryRunQuery(true) match {
-          case Success(_) =>
-            logger.warn("Legacy syntax is deprecated, use SQL syntax instead. " +
-              "See https://cloud.google.com/bigquery/sql-reference/")
-            logger.warn(s"Legacy query: `$sqlQuery`")
-            true
-          case Failure(f) => throw f
+    sqlQuery.trim.split("\n")(0).trim match {
```
Comment (ravwojdyla) [inline scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryClient.scala]: `.toLowerCase` ? probably no reason to be case sensitive here.

**Comment by ravwojdyla** [comment]
Interesting idea. Is there any reason to suspect that someone might use those comments without intent to use this feature? We could make it a bit harder to make that mistake by say using `#syntax=legacysql` (but that is "safety" through obscurity anyway). Alternative might be to make it verbose through the API.

What do you think @nevillelyh @andrewsmartin ?

File: scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryClient.scala (hunk 2 above)
Comment (sinasohangirsc) [inline scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryClient.scala]: Let me fix the case sensitivity here. They indeed are not supposed to be case sensitive per:
https://cloud.google.com/bigquery/docs/reference/standard-sql/enabling-standard-sql

**Comment by sinasohangirsc** [comment]
`#syntax=legacysql` would be a deviation from googles syntax though: 
https://cloud.google.com/bigquery/docs/reference/standard-sql/enabling-standard-sql

**Comment by ravwojdyla** [comment]
@sinasohangirsc oh - i wasn't aware of this feature, this is good to know, in this case 👍

**Comment by andrewsmartin** [comment]
👍

**Comment by ravwojdyla** [comment]
Thanks @sinasohangirsc

**Pull request merged**